### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by manifest**
The PR for the [issue](https://github.com/carllerche/tower-web/issues/28).

### CODE REVIEW

File: src/error/error.rs [hunk 1]
```diff
@@ -75,38 +87,41 @@ impl ErrorKind {
         ErrorKind { kind: BadRequest }
     }
 
-    /// Returns `true` if `self` represents a 400 -- bad request error
-    pub fn is_bad_request(&self) -> bool {
```
Comment (carllerche) [inline src/error/error.rs]: These functions need to stay as they are part of the public API. They can be considered for removal in the next breaking release.

Comment (manifest) [reply src/error/error.rs]: Ok, I see.

**Comment by carllerche** [comment]
Woops, the WIP is gone, it was a relic of my email...

**Comment by manifest** [comment]
Thanks!

**Pull request merged**